spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -221,6 +221,16 @@ class CometSparkSessionExtensions
 
     private def isCometNative(op: SparkPlan): Boolean = op.isInstanceOf[CometNativeExec]
 
+    private def explainChildNotNative(op: SparkPlan): String = {
+      var nonNatives: Seq[String] = Seq()
+      op.children.foreach(plan => {
+        if (!isCometNative(plan)) {
+          nonNatives = nonNatives :+ plan.nodeName
+        }
+      })
```

Comment:
nit: this can be simplified with a functional approach:

```suggestion
      val nonNatives: Seq[String] = op.children.filterNot(isCometNative).map(_.nodeName)
```